src/adaptors/erased.rs

```diff
@@ -13,7 +13,7 @@ use crate::{
 };
 
 /// [`Requester`] with erased type.
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub struct ErasedRequester<'a, E> {
     inner: Arc<dyn ErasableRequester<'a, Err = E> + 'a>,
```

Comment:
This makes the compilation fail (see CI) since `dyn ErasableRequester<'a, Err = E> + 'a` doesn't implement `Debug`. I don't feel good about adding a `Self: Debug` bound to `ErasableRequester`, so you'd need to implement `Debug` by hand or leave `ErasedRequester` baing `!Debug`. If we decide to add an impl (cc @Hirrolot) I think it should use [`finish_non_exhaustive`](https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html#method.finish_non_exhaustive) ie look like this: `ErasedRequester { .. }`.

Reply:
I think we should use `impl` with `finish_non_exhaustive`. This is just the simplest approach for now.